src/mscorlib/src/System/Diagnostics/Eventing/XplatEventLogger.cs

```diff
@@ -109,24 +109,47 @@ private static string Serialize(ReadOnlyCollection<string> payloadName, ReadOnly
                 sb.Append("\\\"");
                 sb.Append(':');
 
-                var valuestr = payload[i] as string;
-
-                if( valuestr != null)
+                Type payloadType = payload[i].GetType();
+                if(payloadType == typeof(string))
+                {
+                    sb.Append("\\\"");
+                    minimalJsonserializer((string)payload[i], sb);
+                    sb.Append("\\\"");
+                }
+                else if(payloadType == typeof(byte[]))
                 {
                     sb.Append("\\\"");
-                    minimalJsonserializer(valuestr,sb);
+                    AppendByteArrayAsHexString(sb, (byte[])payload[i]);
                     sb.Append("\\\"");
                 }
                 else
                 {
                     sb.Append(payload[i].ToString());
                 }
-
             }
             sb.Append('}');
             return StringBuilderCache.GetStringAndRelease(sb);
         }
 
+        private static void AppendByteArrayAsHexString(StringBuilder builder, byte[] byteArray)
+        {
+            if(builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            // Treat NULL byte arrays as empty byte arrays.
+            if(byteArray == null)
+            {
+                byteArray = new byte[0];
+            }
+
+            for(int i=0; i<byteArray.Length; i++)
+            {
+                builder.AppendFormat("{0:X2}", byteArray[i]);
```

Comment:
This is going to allocate a box object for every byte in the byte array.

It'd be cheaper to use the new span-based overloads and do something like:
```C#
ReadOnlySpan<char> hexFormat = "X2";
Span<char> hex = stackalloc char[2];
foreach (byte b in byteArray)
{
    b.TryFormat(hex, out int charsWritten, hexFormat);
    sb.AppendSpan(hex);
}
```